server/src/game/cards/10-experimental/Jealous.ts

```diff
@@ -0,0 +1,56 @@
+import CardType from '@shared/enums/CardType'
+import CardColor from '@shared/enums/CardColor'
+import ServerCard from '../../../models/ServerCard'
+import ServerGame from '../../../models/ServerGame'
+import BuffStrength from '../../../buffs/BuffStrength'
+import BuffDuration from '@shared/enums/BuffDuration'
+import CardFaction from '@shared/enums/CardFaction'
+import CardLocation from '@shared/enums/CardLocation'
+import GameEventType from '@shared/enums/GameEventType'
+import CardFeature from '@shared/enums/CardFeature'
+import ExpansionSet from '@shared/enums/ExpansionSet'
+
+export default class Jealous extends ServerCard {
+	constructor(game: ServerGame) {
+		super(game, {
+			type: CardType.UNIT,
+			color: CardColor.GOLDEN,
+			faction: CardFaction.NEUTRAL,
+			features: [CardFeature.KEYWORD_DEPLOY],
+			stats: {
+				power: 1
+			},
+			expansionSet: ExpansionSet.BASE,
+		})
+
+		this.createDeployEffectTargets()
+			.totalTargets(1)
+			.target(TargetType.CARD_IN_LIBRARY)
+			.requireEnemyUnit()
+
+		this.createEffect(GameEventType.CARD_TARGET_SELECTED_CARD)
+			.perform(({ targetCard }) => this.onTargetSelected(targetCard))
+	}
+
+	private onTargetSelected(target: ServerCard): void {
+		this.game.players.forEach(player => {
+			toDiscardHand = []
+			player.cardHand.unitCards.forEach(card => {
+				if (card.class === target.class) toDiscardHand.push(card)
+			});
+			toDiscardHand.forEach(card => {
+				player.cardHand.discardCard(card)
+			});
```

Comment:
```suggestion
player.cardHand.unitCards.filter(card => card.class === target.class).forEach(card => player.cardHand.discardCard(card))
```
Wouldn't that be the same?

Reply:
not exactly sure on the safety aspects
In some language messing with array while inside for loop is not cool, so I kept it safe
Will fix

Reply:
Filter creates a copy, so it's perfectly fine in any case

Reply:
fixed